float64 offset
bool moving_towards
